### PR TITLE
Prepared address lookup

### PR DESCRIPTION
**Pull request by r4vi**

### CODE REVIEW

**Comment by r4vi** [comment]
I have squashed all your indentation fixed into the original commit

**Comment by r4vi** [comment]
I have rebased on latest master and cleaned up some commits so before you continue to work on this branch I'd suggest deleting your local branch of this and checking out again

File: src/client/styles/components/_form-component.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.message {
+    border-top: 2px solid #ffb81c;
+    background-color: #fff8e8;
```
Comment (teneightfive) [inline src/client/styles/components/_form-component.scss]: It's good practice to use variables for colours. Maybe see if the colours from the [NHS colour palette](https://github.com/nhsuk/frontend/blob/master/src/scss/environment/settings/_colours.scss) can be applied first and if not, create a local colours file for this project in a similar location (`/environment/settings/_colours.scss`).

For reference, this setup is based on the [inverted triangle](https://www.xfive.co/blog/itcss-scalable-maintainable-css-architecture/) approach for structuring CSS. Might be worth a read which will help understand the structure of the files.

File: src/client/styles/components/_form-component.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    color: #825900;
+    padding: 32px;
+}
+.address-message {
+  clear:both;
+}
+.address-message-elem {
```
Comment (teneightfive) [inline src/client/styles/components/_form-component.scss]: Might be worth looking at some of these class names. NHS Frontend and other repos within NHSUK have been following the [BEM](http://getbem.com/introduction/) methodology for naming classes which was also set out in the original organisation [styleguide](https://github.com/nhsuk/styleguides/blob/master/css.md).

File: src/client/styles/base.scss [hunk 3]
```diff
@@ -2,6 +2,7 @@
 @import "nhsuk-frontend/dist/scss/nhsuk";
 
 // components (eg. site search, navigation)
+@import "components/form-component";
```
Comment (teneightfive) [inline src/client/styles/base.scss]: This file name doesn't describe the object defined within it very clearly. Good practice is to name the file after the block (based on BEM) that is defined within that file.

File: src/client/styles/components/_form-component.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  padding-right: 15px;
+  display: inline-block;
+}
```
Comment (teneightfive) [inline src/client/styles/components/_form-component.scss]: Missing new line at the end of the file

File: src/server/plugins/register-form/steps/home-address-lookup-choose.js [hunk 5]
```diff
@@ -25,26 +25,61 @@ function getHandlerFactory(prevSteps) {
...
+              title,
+              prevStep,
+              addressOptions,
+              error: addressError,
+              postcode: postcodeData.postcode,
+              housenumber: postcodeData.houseNumber
+            });
+          });
+        }
+        else{
```
Comment (teneightfive) [inline src/server/plugins/register-form/steps/home-address-lookup-choose.js]: Feels like there might be a few JS style errors that the linter should pick up. Maybe look at addressing things like new space before and after parenthesis and using one true brace style.

**Comment by teneightfive** [comment]
@mtamas5 It's probably worth rebasing and squashing all the fixups before merging this PR so that the history is polluted with fixup commits. [Git fixups](http://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html) are really useful for helping with this.

**Comment by mtamas5** [comment]
I will do that. Also will split up into smaller branches. Just played around with some fixes. I will put you as reviewer on the other one.

**Pull request closed without merging**